pkg/ibm/ibmp.go

```diff
@@ -140,7 +108,60 @@ func (r IBMPowerDynamicConfig) GetInstanceAddress(kubeClient client.Client, ctx
 }
 
 func (r IBMPowerDynamicConfig) ListInstances(kubeClient client.Client, ctx context.Context, instanceTag string) ([]cloud.CloudVMInstance, error) {
-	return nil, fmt.Errorf("not impelemented")
+	log := logr.FromContextOrDiscard(ctx)
+	service, err := r.authenticate(kubeClient, ctx)
+	if err != nil {
+		return nil, err
+	}
+	builder := core.NewRequestBuilder(core.GET).WithContext(ctx)
+	builder.EnableGzipCompression = service.GetEnableGzipCompression()
+
+	pathParamsMap := map[string]string{
+		"cloud": r.pCloudId(),
+	}
+	_, err = builder.ResolveRequestURL(r.Url, `/pcloud/v1/cloud-instances/{cloud}/pvm-instances`, pathParamsMap)
+	if err != nil {
+		return nil, err
+	}
+	builder.AddHeader("CRN", r.CRN)
+	builder.AddHeader("Accept", "application/json")
+
+	request, err := builder.Build()
+	if err != nil {
+		return nil, err
+	}
+
+	var rawResponse map[string]json.RawMessage
+	_, err = service.Request(request, &rawResponse)
+	if err != nil {
+		return nil, err
+	}
+	instancesData := rawResponse["pvmInstances"]
+	var instances []json.RawMessage
+	err = json.Unmarshal(instancesData, &instances)
```

Comment:
can we unmarshal directly into a slice of structures so that the structure of the data we're getting becomes clearer and we can possibly avoid or at lease simplify the translation loop below? 

Actually we probably want to do a little re-factor here to make CloudVMInstance be an interface instead of a plain struct, so that we can have a cloud-specific implementation for it here that we can directly unmarshal into.

Reply:
I am currently experimenting with using 
https://github.com/IBM-Cloud/ppc-aas-go-client/tree/main
so far we can use just the model, and later re-done the whole PPC provider to use it (that's a good piece of work but imo it worth it)